backend/src/v2/driver/k8s_test.go

```diff
@@ -2990,3 +2990,360 @@ func Test_extendPodSpecPatch_PvcMounts_Passthrough_AppliedToPod(t *testing.T) {
+					},
+				},
+			},
+		},
+		{
+			"Valid - capabilities add and drop",
+			&kubernetesplatform.KubernetesExecutorConfig{
+				SecurityContext: &kubernetesplatform.SecurityContext{
+					Capabilities: &kubernetesplatform.Capabilities{
+						Add:  []string{"NET_ADMIN"},
+						Drop: []string{"SYS_ADMIN"},
+					},
+				},
+			},
+			&k8score.PodSpec{
+				Containers: []k8score.Container{
+					{
+						Name: "main",
+						SecurityContext: &k8score.SecurityContext{
+							Capabilities: &k8score.Capabilities{
+								Add:  []k8score.Capability{"NET_ADMIN"},
+								Drop: []k8score.Capability{"SYS_ADMIN"},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			"Valid - SELinux options",
+			&kubernetesplatform.KubernetesExecutorConfig{
+				SecurityContext: &kubernetesplatform.SecurityContext{
+					SeLinuxOptions: &kubernetesplatform.SELinuxOptions{
+						User:  "system_u",
+						Role:  "system_r",
+						Type:  "container_t",
+						Level: "s0:c123,c456",
+					},
+				},
+			},
+			&k8score.PodSpec{
+				Containers: []k8score.Container{
+					{
+						Name: "main",
+						SecurityContext: &k8score.SecurityContext{
+							SELinuxOptions: &k8score.SELinuxOptions{
+								User:  "system_u",
+								Role:  "system_r",
+								Type:  "container_t",
+								Level: "s0:c123,c456",
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			"Valid - seccomp profile RuntimeDefault",
+			&kubernetesplatform.KubernetesExecutorConfig{
+				SecurityContext: &kubernetesplatform.SecurityContext{
+					SeccompProfile: &kubernetesplatform.SeccompProfile{
+						Type: "RuntimeDefault",
+					},
+				},
+			},
+			&k8score.PodSpec{
+				Containers: []k8score.Container{
+					{
+						Name: "main",
+						SecurityContext: &k8score.SecurityContext{
+							SeccompProfile: &k8score.SeccompProfile{
+								Type: k8score.SeccompProfileTypeRuntimeDefault,
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			"Valid - seccomp profile Localhost",
+			&kubernetesplatform.KubernetesExecutorConfig{
+				SecurityContext: &kubernetesplatform.SecurityContext{
+					SeccompProfile: &kubernetesplatform.SeccompProfile{
+						Type:             "Localhost",
+						LocalhostProfile: "profiles/my-profile.json",
+					},
+				},
+			},
+			&k8score.PodSpec{
+				Containers: []k8score.Container{
+					{
+						Name: "main",
+						SecurityContext: &k8score.SecurityContext{
+							SeccompProfile: &k8score.SeccompProfile{
+								Type:             k8score.SeccompProfileTypeLocalhost,
+								LocalhostProfile: func() *string { s := "profiles/my-profile.json"; return &s }(),
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			"Valid - full security context",
+			&kubernetesplatform.KubernetesExecutorConfig{
+				SecurityContext: &kubernetesplatform.SecurityContext{
+					Privileged:               boolPtr(false),
+					AllowPrivilegeEscalation: boolPtr(false),
+					RunAsUser:                int64Ptr(1000),
+					RunAsGroup:               int64Ptr(1000),
```

Comment:
```suggestion
					RunAsGroup:               int64Ptr(0),
```

that is kubernetes best practices for the group